daghelpers.go

```diff
@@ -89,6 +89,26 @@ func GetNodes(ctx context.Context, ds NodeGetter, keys []*cid.Cid) []*NodePromis
 	return promises
 }
 
+func Copy(ctx context.Context, from, to DAGService, root *cid.Cid) error {
+	node, err := from.Get(ctx, root)
+	if err != nil {
+		return err
+	}
+	links := node.Links()
+	if len(links) == 0 {
+		n := node.Copy()
```

Comment:
FYI, you don't actually need to call `Copy` on the node itself. We use that when *editing* nodes (i.e., `Copy` then edit).